src/CommandBinaryBuffer.ts

```diff
@@ -1066,6 +1107,63 @@ class CommandBinaryBuffer extends BinaryBuffer {
         this.setUint16(parameter.value);
     }
 
+    private getParameterBatteryDepassivationInfo (): IParameterBatteryDepassivationInfo {
+        console.log(this.data, this.offset);
+        return {
+            loadTime: this.getUint16(false),
+            internalResistance: this.getUint16(false),
+            lowVoltage: this.getUint16(false)
+        };
+    }
+
+    private setParameterBatteryDepassivationInfo ( parameter: IParameterBatteryDepassivationInfo ): void {
+        this.setUint16(parameter.loadTime, false);
+        this.setUint16(parameter.internalResistance, false);
+        this.setUint16(parameter.lowVoltage, false);
+    }
+
+    private getParameterBatteryMinimalLoadTime (): IParameterBatteryMinimalLoadTime {
+        return {
+            value: this.getUint32(false)
+        };
+    }
+
+    private setParameterBatteryMinimalLoadTime ( parameter: IParameterBatteryMinimalLoadTime ): void {
+        this.setUint32(parameter.value, false);
+    }
+
+    private getParameterSerialNumber (): IParameterSerialNumber {
+        const bytes = [];
+
+        for ( let i = 0; i < SERIAL_NUMBER_SIZE; ++i ) {
+            bytes.push(this.getUint8());
+        }
+
+        return {
+            value: getHexFromBytes(new Uint8Array(bytes))
```

Comment:
```suggestion
            value: getHexFromBytes(this.toUint8Array().slice(...))
```
?

\+ correct offset

Reply:
reworked